source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,131 @@
+exception classes. Сoncrete exception classes are derived from respective STL
+exception classes. oneDAL library shall throw exceptions represented with
+concrete classes.
+
+The root of exception hierarchy is ``onedal::exception``. The rest exception
+classes are derived from this class.
+
+::
+
+    class onedal::exception;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - ``onedal::exception``
+     - Base class of oneDAL exception hierarchy.
+     - Yes
+
+All errors divide into 3 groups: logic errors, runtime errors and bad allocation.
+
+::
+
+    class onedal::logic_error : public onedal::exception;
+    class onedal::runtime_error : public onedal::exception;
+    class onedal::bad_alloc :  public onedal::exception, public std::bad_alloc;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - ``onedal::logic_error``
+     - Reports violations of preconditions and invariants.
+     - Yes
+   * - ``onedal::runtime_error``
+     - Reports violations of postconditions and other errors happened during the
+       execution of oneDAL functionality.
+     - Yes
+   * - ``onedal::bad_alloc``
+     - Reports failure to allocate storage.
+     - No
+
+Precondition and invariant errors that are represented by
+``onedal::logic_error`` are divided into subgroups: invalid argument, domain
+error, out of range error, unimplemented method or algorithm, unavailable
+device or data.
+
+::
+
+    class onedal::invalid_argument :  public onedal::logic_error, public std::invalid_argument;
+    class onedal::domain_error :  public onedal::logic_error, public std::domain_error;
+    class onedal::out_of_range :  public onedal::logic_error, public std::out_of_range;
+    class onedal::unimplemented_error :  public onedal::logic_error, public std::logic_error;
+    class onedal::unavailable_error :  public onedal::logic_error, public std::logic_error;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - ``onedal::invalid_argument``
+     - Reports situations when the argument was not been accepted.
+     - No
+   * - ``onedal::domain_error``
+     - Reports situations when the argument is outside of the domain on which
+       the operation is defined. Higher priority than
+       ``onedal::invalid_argument``.
+     - No
+   * - ``onedal::out_of_range``
+     - Reports situations when the index is out of range. Higher priority
+       than ``onedal::invalid_argument``.
+     - No
+   * - ``onedal::unimplemented_error``
+     - Reports errors that arise because an algorithm or a method is not
+       implemented.
+     - No
+   * - ``onedal::unavailable_error``
+     - Reports situations when a device or data is not available.
+     - No
+
+If an error occurs during function execution after preconditions and invariants
```

Comment:
Align wording with the style above
Errors that occurs during the execution of toneDAL functionality are represented with  ... shall be divided into...